x-pack/plugins/spaces/server/lib/spaces_client.ts

```diff
@@ -195,6 +196,19 @@ export class SpacesClient {
     await repository.deleteByNamespace(id);
   }
 
+  public async canManageSavedObjects(spaceId: string): Promise<boolean> {
```

Comment:
This doesn't really belong in the `SpacesClient`, but I put it here for a couple of reasons:
1) We have similar logic [here](https://github.com/elastic/kibana/pull/38014/files#diff-a956a5b5b7800bb09089b946a4b93573R24) to determine if a user is allowed to enumerate spaces.
2) It keeps the spaces -> authorization coupling contained to the `SpacesClient`, which will hopefully make the authz extraction easier when we remove the circular dependency between security and spaces